agent/agent.go

```diff
@@ -447,6 +447,17 @@ func stopServiceInputs(inputs []*models.RunningInput) {
 	}
 }
 
+// stopRunningOutputs stops all running outputs.
+func stopRunningOutputs(outputs []*models.RunningOutput) {
+	for _, output := range outputs {
```

Comment:
You might want to close the outputs in parallel!?

Reply:
I don't think that's necessary.